packages/suite/src/hooks/suite/useBioAuthDesktopApi.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { isMacOs } from '@trezor/env-utils';
+import { desktopApi } from '@trezor/suite-desktop-api';
+
+import {
+    requestBioAuthChangeThunk,
+    requestBioAuthValidationThunk,
+} from 'src/actions/suite/bioAuthThunks';
+import { selectBioAuthEnabled } from 'src/reducers/bioAuth';
+
+import { useDispatch } from './useDispatch';
+import { useSelector } from './useSelector';
+import { useTranslation } from './useTranslation';
+
+export const useBioAuthDesktopApi = () => {
```

Comment:
I am in general against using hooks, they make the code look like undebuggable spaghetti and put the business logic where it shouldn't be IMHO.

There's a myriad of articles about it https://leocode.com/development/why-hooks-are-bad/


I have made a notion article explaining what could be a better approach to it https://www.notion.so/satoshilabs/Trezor-Suite-Architecture-improvements-19ddc526060680d6af54cc653423e25d#19ddc526060680d6af54cc653423e25d

I am also nothing tha this very important piece of code has not a line of test and is pretty hard to test...